spec/models/comment_spec.rb

```diff
@@ -28,14 +33,55 @@
     end
 
     it { is_expected.to validate_length_of(:body_markdown).is_at_least(1).is_at_most(25_000) }
-    it { is_expected.to validate_inclusion_of(:commentable_type).in_array(%w[Article PodcastEpisode]) }
 
-    it "is invalid if commentable is unpublished article" do
-      # rubocop:disable RSpec/NamedSubject
-      subject.commentable = build(:article, published: false)
-      expect(subject).not_to be_valid
-      # rubocop:enable RSpec/NamedSubject
+    # rubocop:disable RSpec/NamedSubject
+    describe "commentable" do
+      it "is invalid if commentable is an unpublished article" do
```

Comment:
Interestingly, I just found out that if someone unpublishes a post then you can no longer view those comments through the permalink. However, we do allow comments to be viewed from deleted posts. Wonder if this is something we should change. 🤔

Reply:
> Let me know if you need help with the notifications. Also if you give me a heads up before this goes out I could increase our STATEMENT_TIMEOUT to try and get the job to just do it all successfully.

Sure, let's sync up tomorrow UTC afternoon

> Interestingly, I just found out that if someone unpublishes a post then you can no longer view those comments through the permalink. However, we do allow comments to be viewed from deleted posts. Wonder if this is something we should change. 🤔

I don't know TBH, it's probably more a product decision. In a way it would be odd to look at a comment belonging to something that has been unpublished, but on the other side as you mentioned, we keep comments for deleted articles because they intrinsically belong to the commenter, not the article owner. 

There's a catch we haven't explored in either situation: what if a comment quotes a part of the article and the article owner doesn't want its content to be indexed anymore? Thus the unpublish or the delete. Food for thought